tests/application_development/code_relocation fails to build for mr_canhubk3

**Describe the bug**
`tests/application_development/code_relocation/tests.application_development.code_relocation` fails to build for `mr_canhubk3`.

**To Reproduce**
`west build -p always -b mr_canhubk3 tests/application_development/code_relocation -T tests.application_development.code_relocation`

**Expected behavior**
Test can build and run successfully on this board.

**Impact**
May provoke CI failures in unrelated PR's, and making failed builds in:
https://github.com/zephyrproject-rtos/zephyr/pull/60115
https://github.com/zephyrproject-rtos/zephyr/pull/60159
https://github.com/zephyrproject-rtos/zephyr/pull/60162

**Logs and console output**
```
[195/200] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /__w/zephyr/zephyr/twister-out/mr_canhubk3/tests/application_development/code_relocation/tests.application_development.code_relocation/zephyr/zephyr_pre0.map 
: && ccache /opt/toolchains/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc  -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -fuse-ld=bfd  -T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/__w/zephyr/zephyr/twister-out/mr_canhubk3/tests/application_development/code_relocation/tests.application_development.code_relocation/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a  zephyr/arch/arch/arm/core/aarch32/mpu/libarch__arm__core__aarch32__mpu.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/libc/common/liblib__libc__common.a  zephyr/soc/arm/nxp_s32/s32k/libsoc__arm__nxp_s32__s32k.a  zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/hal_nxp/hal_nxp/lib..__modules__hal__nxp.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/opt/toolchains/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/thumb/v7e-m+fp/hard"  -L/__w/zephyr/zephyr/twister-out/mr_canhubk3/tests/application_development/code_relocation/tests.application_development.code_relocation/zephyr  -lgcc  zephyr/libcode_relocation_source_lib.a  test_lib/libtest_lib.a  zephyr/arch/common/libisr_tables.a  -Wl,--fatal-warnings  -mcpu=cortex-m7  -mthumb  -mabi=aapcs  -mfpu=fpv5-sp-d16  -mfloat-abi=hard  -mfp16-format=ieee  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,-no-pie && cd /__w/zephyr/zephyr/twister-out/mr_canhubk3/tests/application_development/code_relocation/tests.application_development.code_relocation/zephyr && /usr/bin/cmake -E true
/opt/toolchains/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd:zephyr/linker_zephyr_pre0.cmd:451: warning: memory region `IVT_HEADER' not declared
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Linux, Windows
 - Toolchain Zephyr SDK 0.16.1
 - Commit SHA 5f0bba7ced041c791561fc49c4af12566f8a8f83

**Additional context**
It seems like this test wasn't build by CI in the PR were this board was introduced, so not caught earlier.

--- a/tests/application_development/code_relocation/linker_arm_sram2.ld
+++ b/tests/application_development/code_relocation/linker_arm_sram2.ld
@@ -36,4 +36,5 @@ MEMORY
 #endif
     }
 
-#include <zephyr/arch/arm/aarch32/cortex_m/scripts/linker.ld>
+/* Include the SoC-specific linker file */
+#include <linker.ld>

--- a/tests/application_development/code_relocation/testcase.yaml
+++ b/tests/application_development/code_relocation/testcase.yaml
@@ -6,12 +6,15 @@ tests:
     arch_allow: arm
     extra_configs:
       - CONFIG_RELOCATE_TO_ITCM=y
+    platform_allow:
+      - mimxrt1060_evk
   tests.application_development.code_relocation_kinetis:
-    filter: CONFIG_CPU_HAS_NXP_MPU and CONFIG_MINIMAL_LIBC and dt_chosen_enabled("zephyr,itcm")
+    filter: CONFIG_CPU_HAS_NXP_MPU and CONFIG_MINIMAL_LIBC
     arch_allow: arm
     extra_configs:
-      - CONFIG_RELOCATE_TO_ITCM=y
       - CONFIG_MPU_ALLOW_FLASH_WRITE=y
+    platform_allow:
+      - frdm_k64f
   tests.application_development.code_relocation.no_itcm:
     filter: not CONFIG_CPU_HAS_NXP_MPU and not dt_chosen_enabled("zephyr,itcm")
     arch_allow: arm

--- a/tests/application_development/code_relocation/testcase.yaml
+++ b/tests/application_development/code_relocation/testcase.yaml
@@ -15,6 +15,14 @@ tests:
       - CONFIG_MPU_ALLOW_FLASH_WRITE=y
     platform_allow:
       - frdm_k64f
+  tests.application_development.code_relocation.nxp_s32:
+    filter: not CONFIG_CPU_HAS_NXP_MPU and CONFIG_MINIMAL_LIBC and dt_chosen_enabled("zephyr,itcm")
+    arch_allow: arm
+    extra_configs:
+      - CONFIG_RELOCATE_TO_ITCM=y
+      - CONFIG_NULL_POINTER_EXCEPTION_DETECTION_NONE=y
+    platform_allow:
+      - mr_canhubk3
   tests.application_development.code_relocation.no_itcm:
     filter: not CONFIG_CPU_HAS_NXP_MPU and not dt_chosen_enabled("zephyr,itcm")
     arch_allow: arm

